contracts/allowlist/ValidatorSmartContractAllowList.sol

```diff
@@ -0,0 +1,127 @@
+    constructor (address[] memory initialAccounts, address[] memory initialValidators) public {
+        require(initialAccounts.length > 0, "no inital allowed accounts");
+        require(initialValidators.length > 0, "no inital validator accounts");
+
+        require(initialAccounts.length >= initialValidators.length, "number of initial accounts smaller than number of inital validators");
+        for (uint i = 0; i < initialAccounts.length; i++) {
+            require(initialAccounts[i] != address(0), "initial accounts cannot be zero");
+            numAllowedAccounts++;
+            if (i < initialValidators.length) {
+                require(initialValidators[i] != address(0), "initial validators cannot be zero");
+                allowedAccounts[initialAccounts[i]] = accountInfo(true, true, uint8(i));
+                validators.push(initialValidators[i]);
+            } else {
+                allowedAccounts[initialAccounts[i]] = accountInfo(true, false, 0);
+            }
+        }
+    }
+
+    function getValidators() override external view returns (address[] memory) {
+        return validators;
+    }
+
+    function activate(address newValidator) external senderIsAllowed {
+        require(newValidator != address(0), "cannot activate validator with address 0");
+        uint i;
+        for (i=0; i < validators.length; i++) {
+            require(newValidator != validators[i], "validator is already active");
+        }
+        if (allowedAccounts[msg.sender].activeValidator) {
+            validators[allowedAccounts[msg.sender].validatorIndex] = newValidator;
+        } else {
+            allowedAccounts[msg.sender] = accountInfo(true, true, uint8(validators.length));
+            validators.push(newValidator);
+        }
+
+    }
+
+    function deactivate() external senderIsAllowed {
+        require(validators.length > 1, "cannot deactivate last validator");
+        require(allowedAccounts[msg.sender].activeValidator, "sender does not have an active validator");
+        allowedAccounts[msg.sender].activeValidator = false;
+        validators[allowedAccounts[msg.sender].validatorIndex] = validators[validators.length-1];
+        validators.pop();
+    }
+
+    function voteToAddAccountToAllowList(address account) external senderIsAllowed {
+        require(account != address(0), "account to be added cannot be 0");
+        require(allowedAccounts[account].allowed == false, "account to add is already on the allow list");
+
+        for (uint i=0; i < currentVotes[account].length; i++) {
+            require(currentVotes[account][i] != msg.sender, "sender has already voted to add account");
+        }
+        currentVotes[account].push(msg.sender);
+    }
+
+    function voteToRemoveAccountFromAllowList(address account) external senderIsAllowed {
+        require(account != address(0), "account to be added cannot be 0");
+        require(allowedAccounts[account].allowed == true, "account to remove is not on the allow list");
```

Comment:
Except for this `require` both `voteToAddAccountToAllowList` and `voteToRemoveAccountFromAllowList` do exactly the same. Maybe factor out their common tail into an internal function? Or just have a single public function with a `bool addOrRemove` parameter.